tools/protovalidate-conformance/internal/cases/cases_oneof.go

```diff
@@ -59,10 +60,22 @@ func oneofSuite() suites.Suite {
 			Message:  &cases.Oneof{O: &cases.Oneof_Z{Z: &cases.TestOneofMsg{}}},
 			Expected: results.Violations(&validate.Violation{FieldPath: "z.val", ConstraintId: "bool.const"}),
 		},
-		"required/valid": {
+		"required/valid_scalar": {
+			Message:  &cases.OneofRequired{O: &cases.OneofRequired_X{X: "foo"}},
+			Expected: results.Success(true),
+		},
+		"required/empty_scalar": {
 			Message:  &cases.OneofRequired{O: &cases.OneofRequired_X{X: ""}},
+			Expected: results.Violations(&validate.Violation{FieldPath: "o", ConstraintId: "required"}),
+		},
+		"required/valid_message": {
+			Message:  &cases.OneofRequired{O: &cases.OneofRequired_Duration{Duration: &durationpb.Duration{}}},
 			Expected: results.Success(true),
 		},
+		"required/empty_message": {
+			Message:  &cases.OneofRequired{O: &cases.OneofRequired_Duration{Duration: nil}},
+			Expected: results.Violations(&validate.Violation{FieldPath: "o", ConstraintId: "required"}),
```

Comment:
I _think_ this should fail `(buf.validate.oneof).required`. There is no parallel in a regular (non-oneof) field. Setting `Duration` to `nil` here seems like an invalid state, but we still need _some_ behavior from protovalidate.